cloudplatform/connectivity-oauth/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/IdentityAuthenticationServiceBindingDestinationLoader.java

```diff
@@ -0,0 +1,142 @@
+    @Nonnull
+    private final ServiceBindingDestinationLoader delegateLoader;
+
+    public IdentityAuthenticationServiceBindingDestinationLoader()
+    {
+        this(null);
+    }
+
+    // for testing purposes
+    IdentityAuthenticationServiceBindingDestinationLoader(
+        @Nullable final ServiceBindingDestinationLoader delegateLoader )
+    {
+        this.delegateLoader =
+            delegateLoader != null ? delegateLoader : ServiceBindingDestinationLoader.defaultLoaderChain();
+    }
+
+    @Nonnull
+    @Override
+    public Try<HttpDestination> tryGetDestination( @Nonnull final ServiceBindingDestinationOptions options )
+    {
+        final ServiceBinding serviceBinding = options.getServiceBinding();
+        if( !isIdentityAuthenticationServiceBinding(serviceBinding) ) {
+            return Try.failure(new DestinationNotFoundException("The bound service is not backed by the IAS service."));
+        }
+
+        final List<EndpointEntry> endpoints = EndpointEntry.allFromServiceBinding(serviceBinding);
+        final EndpointEntry endpoint = getEndpoint(options, endpoints);
+        if( endpoint == null ) {
+            final DestinationAccessException exception =
+                new DestinationAccessException("Unable to determine the endpoint for the IAS-based service binding.");
+            return Try.failure(exception);
+        }
+
+        final ServiceBindingDestinationOptions.Builder optionsBuilder =
+            ServiceBindingDestinationOptions
+                .forService(ServiceIdentifier.of("identity"))
+                .onBehalfOf(options.getOnBehalfOf())
+                .withOption(BtpServiceOptions.IasOptions.withTargetUri(endpoint.uri));
+
+        if( endpoint.mTLSOnly ) {
+            optionsBuilder.withOption(BtpServiceOptions.IasOptions.withMTLSAuthenticationOnly());
+        }
+
+        return delegateLoader.tryGetDestination(optionsBuilder.build());
+    }
+
+    private static boolean isIdentityAuthenticationServiceBinding( @Nonnull final ServiceBinding serviceBinding )
+    {
+        final Object rawAuthenticationService = serviceBinding.getCredentials().get("authentication-service");
+        if( !(rawAuthenticationService instanceof String) ) {
+            return false;
+        }
+
+        return "identity".equalsIgnoreCase((String) rawAuthenticationService);
+    }
+
+    @Nullable
+    private static EndpointEntry getEndpoint(
+        @Nonnull final ServiceBindingDestinationOptions options,
+        @Nonnull final List<EndpointEntry> allEndpoints )
+    {
+        // TODO: implement a generic option so that users can select the endpoint they want to use
+        if( allEndpoints.size() != 1 ) {
+            log.warn("IAS-based service binding does not contain exactly one endpoint.");
+            return null;
+        }
+
+        return allEndpoints.get(0);
+    }
+
+    @Value
+    private static class EndpointEntry
+    {
+        String name;
+        URI uri;
+        boolean mTLSOnly;
```

Comment:
(Major-ly)

I also noticed this in other (recently added) code. Camel-case also applies to abbreviations in our code-base:
```suggestion
        boolean mTlsOnly;
```

It's a nitpick, but I like the consistency :)

Reply:
Agreed.

I used your remark to also refactor our existing (but not yet communicated) APIs:

```diff
- IasOptions.withMTLSOnlyAuthentication()
+ IasOptions.withMutualTlsOnly()
```

Personally, I felt like spelling our the **m** somewhat looks better.
Any opinions on those changes?

Reply:
> I felt like spelling out the m somewhat looks better.

I agree.. though not a strong opinion 😁